cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandlerV2.java

```diff
@@ -0,0 +1,113 @@
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * The {@link HttpHandler} for handling REST call to stream endpoints.
+ *
+ * TODO: Currently stream "dataset" is implementing old dataset API, hence not supporting multi-tenancy.
+ */
+@Path(Constants.Gateway.API_VERSION_2 + "/streams")
+public final class StreamHandlerV2 extends AuthenticatedHttpHandler {
+
+  // StreamHandler for V3 APIs,to which calls will be delegated to.
+  private final StreamHandler streamHandler;
+
+  @Inject
+  public StreamHandlerV2(Authenticator authenticator, StreamHandler streamHandler) {
+    super(authenticator);
+    this.streamHandler = streamHandler;
+  }
+
+  @GET
+  @Path("/{stream}/info")
+  public void getInfo(HttpRequest request, HttpResponder responder,
+                      @PathParam("stream") String stream) throws Exception {
+    streamHandler.getInfo(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, stream);
+  }
+
+  @PUT
+  @Path("/{stream}")
+  public void create(HttpRequest request, HttpResponder responder,
+                     @PathParam("stream") String stream) throws Exception {
+    streamHandler.create(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, stream);
+  }
+
+  @POST
+  @Path("/{stream}")
+  public void enqueue(HttpRequest request, HttpResponder responder,
+                      @PathParam("stream") String stream) throws Exception {
+    streamHandler.enqueue(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, stream);
+  }
+
+  @POST
+  @Path("/{stream}/async")
+  public void asyncEnqueue(HttpRequest request, HttpResponder responder,
+                           @PathParam("stream") String stream) throws Exception {
+    streamHandler.asyncEnqueue(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, stream);
+  }
+
+  @POST
+  @Path("/{stream}/batch")
+  public BodyConsumer batch(HttpRequest request, HttpResponder responder,
+                            @PathParam("stream") String stream) throws Exception {
+    return streamHandler.batch(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, stream);
+  }
+
+  @POST
+  @Path("/{stream}/truncate")
+  public void truncate(HttpRequest request, HttpResponder responder,
+                       @PathParam("stream") String stream) throws Exception {
+    streamHandler.truncate(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, stream);
+  }
+
+  @PUT
+  @Path("/{stream}/config")
+  public void setConfig(HttpRequest request, HttpResponder responder,
+                        @PathParam("stream") String stream) throws Exception {
+    streamHandler.setConfig(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, stream);
+  }
+
+  /**
+   * Updates the request URI to its v3 URI before delegating the call to the corresponding v3 handler.
+   * Note: This piece of code is duplicated in LogHandler, but its ok since this temporary, till we
```

Comment:
nit: comment is inappropriate - I know its copied, but its not just in ``LogHandler``